frontend/integration-tests/views/sidenav.view.ts

```diff
@@ -1,10 +1,14 @@
 import { $$, by, browser, element, ExpectedConditions as until } from 'protractor';
 
-export const navSectionFor = (name: string) => element(by.cssContainingText('.pf-c-nav > .pf-c-nav__list > .pf-c-nav__item', name));
+export const navSectionFor = (name: string) => element(by.linkText(name));
 
 export const clickNavLink = (path: [string, string]) => browser.wait(until.visibilityOf(navSectionFor(path[0])))
-  .then(() => navSectionFor(path[0]).click())
-  .then(() => browser.wait(until.visibilityOf(navSectionFor(path[0]).element(by.linkText(path[1])))))
-  .then(() => navSectionFor(path[0]).element(by.linkText(path[1])).click());
+  .then(() => navSectionFor(path[1]).isDisplayed().then(function (isVisible) {
+    if (!isVisible) {
+        navSectionFor(path[0]).click();
```

Comment:
You need to return a new promise here.

Reply:
I am not sure how to that but AFAIK, a call to promise.then returns a promise, that s why chaining is possible

Reply:
This function does not wait for anything inside the `then()` to happen. It will continue before `navSectionFor` and `browser.wait` execute. The promise handling is still incorrect here.